File: 1/1.py
#!/usr/bin/env python
key = {"(": 1, ")": -1}
level = index = 0
for dir in raw_input():
    index += 1
    level += key[dir]
    if level == -1:
        print "Basement position: " + str(index)
print "Level: " + str(level)

File: 1/1.rb
#!/usr/bin/env ruby

level = 0
key = {"(" => 1, ")" => -1}
basement = nil

gets.chomp.each_char.with_index(1) do |dir, index|
  puts "#{dir} #{index}"
  level += key[dir]
  basement = index if level == -1 && !basement
end

puts "Final Level: #{level}"
puts "Reached basement: #{basement}" if basement
